rclpy/test/test_clock.py

```diff
@@ -211,6 +216,137 @@ def test_triggered_clock_change_callbacks(self):
         handler3.unregister()
 
 
+@pytest.fixture()
+def default_context():
+    rclpy.init()
+    yield get_default_context()
+    rclpy.shutdown()
+
+
+@pytest.fixture()
+def non_default_context():
+    context = Context()
+    context.init()
+    yield context
+    context.try_shutdown()
+
+
+def test_sleep_until_mismatched_clock_type(default_context):
+    clock = Clock(clock_type=ClockType.SYSTEM_TIME)
+    with pytest.raises(ValueError, match='.*clock type does not match.*'):
+        clock.sleep_until(Time(clock_type=ClockType.STEADY_TIME))
+
+
+def test_sleep_until_non_default_context(non_default_context):
+    clock = Clock()
+    assert clock.sleep_until(clock.now() + Duration(seconds=0.1), context=non_default_context)
```

Comment:
Is there a chance that we'll have spurious wakeups in a 100 millisecond wait (especially on Windows and macOS)?  That is, is this a potentially flaky test?

(the same question goes for all of the places where we actually sleep below)

Reply:
> Is the question: will the underlying std::condition_variable::wait spuriously wake up and cause sleep_until() to return False?

Yeah, that is basically the question.

> The rclpy::Event::wait_...() are using the predicate version of std::condition_variable::wait() and std::condition_variable::wait_until() to prevent spurious wake ups.

Ah right.  OK, in that case this should be fine.  Thanks for the explanation.